kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverVariantConfig.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2017-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.kover.gradle.plugin.dsl
+
+import org.gradle.api.Action
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+
+
+public interface KoverVariantsRootConfig: KoverVariantConfig {
+    public fun create(variantName: String, block: Action<KoverVariantCreateConfig>)
+
+    public fun provided(variantName: String, block: Action<KoverVariantConfig>)
+
+    public fun total(block: Action<KoverVariantConfig>)
+}
+
+public interface KoverVariantConfig {
+    public fun classes(block: Action<KoverVariantClasses>)
+    public fun instrumentation(block: Action<KoverVariantInstrumentation>)
+    public fun testTasks(block: Action<KoverVariantTestTasks>)
+}
+
+public interface KoverVariantClasses {
+    public val excludeJava: Property<Boolean>
+    public val excludedSourceSets: SetProperty<String>
+}
+
+public interface KoverVariantInstrumentation {
+    public val disableAll: Property<Boolean>
+    public val excludedClasses: SetProperty<String>
```

Comment:
Instrumentation settings should be a very, very rarely used feature - it is needed when a serious class modification error occurs and a specific class needs to be excluded.

If you leave the setting in this form 
```
kover {
    variants {
        classes {
            excludedClasses("Foo")
        }
    }
}
```
then it will be easy to get confused about what `classes {}` is doing - excluding classes from the report or something else.


The purpose of the `classes` block is to exclude a large group of classes from sharing by some sign (source set, language) - and as a result from all reports.
`instrumentation` controls which classes won't be modified, but they will still be included in the report, although with 0% coverage.

Perhaps naming `classes` is not the most successful.

Reply:
But `classes { sourceSets(...) }` will exclude them from instrumentation too, won't they?

I agree that `classes` is not the most understandable name

Reply:
> But classes { sourceSets(...) } will exclude them from instrumentation too, won't they?

That's the thing, no, otherwise it can break the measurement of inline functions.

This is only guaranteed to exclude classes grouped by source set from all reports

Reply:
Maybe it should be in reports filtering then?

Reply:
Reports are an aggregation block for all projects from dependencies. The filters declared in it are not exported to dependent projects.

Here, the setting regulates that classes from the specified source sets are always excluded from a specific project and in all reports

Reply:
This is an important difference that should be mentioned in the documentation